reframe/core/variables.py

```diff
@@ -172,16 +172,40 @@ class MyRequiredTest(HelloTest):
         the field validator.
     :returns: A new test variable.
 
-    .. versionadded:: 3.10.2
+    .. version added:: 3.10.2
        The ``loggable`` argument is added.
 
     '''
 
-    __slots__ = ('_default_value', '_field', '_loggable', '_name')
+    # NOTE: We can't use truly private fields in __slots__, because
+    # __setattr__() will be called with their mangled name and we cannot match
+    # them in the __slots__ with making implementation-defined assumptions
+    # about the mangled name. So we just add the `_p_` prefix for "private"
+
+    __slots__ = ('_p_default_value', '_p_field',
+                 '_loggable', '_name', '_target')
+
+    __mutable_props = ('_default_value',)
 
     def __init__(self, *args, **kwargs):
         field_type = kwargs.pop('field', fields.TypedField)
-        self._default_value = kwargs.pop('value', Undefined)
+        alias = kwargs.pop('alias', None)
+
+        if alias and not isinstance(alias, TestVar):
```

Comment:
Does it make sense to move these tests to a function?

Reply:
Why? They are only used here.